lib/index-basic.js

```diff
@@ -1,10 +1,16 @@
 'use strict';
 
-var Plotly = require('./core');
+var core = require('../src/core');
 
-Plotly.register([
-    require('./bar'),
-    require('./pie')
+core.register([
+    require('../src/traces/bar'),
+    require('../src/traces/pie'),
+
+    require('../src/transforms/aggregate'),
+    require('../src/transforms/filter'),
+    require('../src/transforms/groupby'),
+    require('../src/transforms/sort'),
+    require('../src/components/calendars')
 ]);
 
-module.exports = require('./register_extra')(Plotly);
+module.exports = (function(Plotly) { return Plotly; })(core);
```

Comment:
Personally it feels to me like this indirection makes things more confusing than the previous pattern - which with `register_extra` removed becomes just:
```js
var Plotly = require('../src/core');

Plotly.register([...]);

module.exports = Plotly;
```

Is there another reason for it?

Reply:
`register_extra` file is not documented and confusing.
Until we could bump d3 (cc: #5112) or possibly release a `@plotly/d3-v3` patch to fix unexpected characters at build time we need a function to the return correct result.